batch/create-account/create-account-user-subscription.sh

```diff
@@ -1,40 +1,89 @@
 #!/bin/bash
+# Failed validation in Cloud Shell on 4/7/2022
+
+# <FullScript>
+# Create a Batch account in user subscription mode
+
+# Variable block
+let "randomIdentifier=$RANDOM*$RANDOM"
+subscriptionId="$(az account show --query id -o tsv)"
+role="contributor"
+location="East US"
+servicePrincipal="msdocssp$randomIdentifier"
+resourceGroup="msdocs-batch-rg-$randomIdentifier"
+tag="create-account"
+keyVault="msdocskeyvault$randomIdentifier"
+batchAccount="msdocsbatch$randomIdentifier"
+storageAccount="msdocsstorage$randomIdentifier"
+skuStorage="Standard_LRS"
 
 # Allow Azure Batch to access the subscription (one-time operation).
-az role assignment create \
-    --assignee  ddbf3205-c6bd-46ae-8127-60eb93363864 \
-    --role contributor
+#az role assignment create \
+#    --assignee  ddbf3205-c6bd-46ae-8127-60eb93363864 \
+#    --role contributor
 
-# Create a resource group.
-az group create --name myResourceGroup --location westeurope
+# Create a resource group
+echo "Creating $resourceGroup in "$location"..."
+az group create --name $resourceGroup --location "$location" --tag $tag
+
+# Create the service principal with rights scoped to the registry.
+# Default permissions are for docker pull access. Modify the '--role'
+# argument value as desired:
+# acrpull:     pull only
+# acrpush:     push and pull
+# owner:       push, pull, and assign roles
+
+az ad sp create-for-rbac --name $servicePrincipal --role contributor  --scopes /subscriptions/$subscriptionId/resourceGroups/$resourceGroup
+apId=$(az ad sp list --display-name $servicePrincipal --query "[].appId" --output tsv)
+
+az role assignment create \
+   --assignee  $apId \
+   --role contributor
 
 # Create an Azure Key Vault. A Batch account that allocates pools in the user's subscription 
-# must be configured with a Key Vault located in the same region. 
+# must be configured with a Key Vault located in the same region.
+echo "Creating $keyVault" 
 az keyvault create \
-    --resource-group myResourceGroup \
-    --name mykevault \
-    --location westeurope \
+    --resource-group $resourceGroup \
+    --name $keyVault \
+    --location "$location" \
     --enabled-for-deployment true \
     --enabled-for-disk-encryption true \
     --enabled-for-template-deployment true
 
 # Add an access policy to the Key Vault to allow access by the Batch Service.
 az keyvault set-policy \
-    --resource-group myResourceGroup \
-    --name mykevault \
-    --spn ddbf3205-c6bd-46ae-8127-60eb93363864 \
```

Comment:
If you are creating a Batch account in User subscription mode, when you link the key vault, you have to specify the Microsoft Azure Batch (Application ID ddbf3205-c6bd-46ae-8127-60eb93363864) as the Service Principal. So specifying ddbf3205-c6bd-46ae-8127-60eb93363864 originally for the spn parameter should be what allows the Batch account to access the key vault.